packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -1,21 +1,246 @@
+    if (!active) return
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_HOVER`,
+      orgId,
+      siteId,
+      buildId,
+      name: `link hover`,
+    })
+  }
+
+  return (
+    <IndicatorButton
+      testId={`link`}
+      iconSvg={linkIcon}
+      onClick={copyLinkClick}
+      onMouseEnter={trackHover}
+      {...props}
+      {...linkButtonCopyProps}
+    />
+  )
+}
+
+export function InfoIndicatorButton(props) {
+  const { active, orgId, siteId, buildId } = props
+  const trackHover = () => {
+    if (!active) return
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_HOVER`,
+      orgId,
+      siteId,
+      buildId,
+      name: `info hover`,
+    })
+  }
+  return (
+    <IndicatorButton
+      testId="info"
+      iconSvg={infoIcon}
+      onMouseEnter={trackHover}
+      {...props}
+    />
+  )
+}
+
+export function BuildErrorIndicatorTooltip({ siteId, orgId, buildId }) {
+  return (
+    <>
+      {failedIcon}
+      {`Unable to build preview`}
+      <a
+        href={generateBuildLogUrl({ orgId, siteId, buildId })}
+        target="_blank"
+        rel="noreferrer"
+        onClick={() => {
+          trackEvent({
+            eventType: `PREVIEW_INDICATOR_CLICK`,
+            orgId,
+            siteId,
+            buildId,
+            name: `error logs`,
+          })
+        }}
+        data-gatsby-preview-indicator="tooltip-link"
+      >
+        <p data-gatsby-preview-indicator="tooltip-link-text">{`View logs`}</p>
+        <div data-gatsby-preview-indicator="tooltip-svg">{logsIcon}</div>
+      </a>
+    </>
+  )
+}
+
+function BuildSuccessIndicatorTooltip({
+  isOnPrettyUrl,
+  sitePrefix,
+  orgId,
+  siteId,
+  buildId,
+}) {
+  return (
+    <>
+      {`New preview available`}
+      <a
+        onClick={() => {
+          newPreviewAvailableClick({
+            isOnPrettyUrl,
+            sitePrefix,
+            orgId,
+            siteId,
+            buildId,
+          })
+        }}
+        data-gatsby-preview-indicator="tooltip-link"
+      >
+        <p data-gatsby-preview-indicator="tooltip-link-text">{`Click to view`}</p>
+      </a>
+    </>
+  )
+}
+
+const generateBuildLogUrl = ({ orgId, siteId, buildId }) => {
+  const pathToBuildLogs = `https://www.gatsbyjs.com/dashboard/${orgId}/sites/${siteId}/builds/${buildId}/details`
+  const returnTo = encodeURIComponent(pathToBuildLogs)
+
+  return `${pathToBuildLogs}?returnTo=${returnTo}`
+}
+
+const delay = ms => new Promise(resolve => setTimeout(resolve, ms || 50))
+
+const newPreviewAvailableClick = async ({
+  isOnPrettyUrl,
+  sitePrefix,
+  orgId,
+  siteId,
+  buildId,
+}) => {
+  trackEvent({
+    eventType: `PREVIEW_INDICATOR_CLICK`,
+    orgId,
+    siteId,
+    buildId,
+    name: `new preview`,
+  })
+
+  /**
+   * Delay to ensure that track event fires but do not await trackEvent directly since we do not
+   * want to block the thread until the event request comes back
+   */
+  await delay(75)
+
+  if (!window) {
```

Comment:
When is window undefined? It's happening on a click which can't happen in SSR mode.

Reply:
Ehh, I don't remember why this is here. Maybe just brought it over with the refactor. Gonna remove it